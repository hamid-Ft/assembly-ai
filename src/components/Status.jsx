// eslint-disable-next-line react/prop-types
const Status = ({ isLoading, status }) => {
  return (
    <div>
      <p>
        {isLoading
          ? `Calculating...${status || "uploading "}...`
          : "Give me audio!"}
      </p>
    </div>
  );
};

export default Status;
