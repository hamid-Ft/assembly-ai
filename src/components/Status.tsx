import React from "react";

interface StatusProps {
  isLoading: boolean;
  status: string;
}

const Status = ({ isLoading, status }: StatusProps) => {
  return (
    <div>
      <p>
        {isLoading
          ? `Calculating...${status || "uploading "}...`
          : "Give me audio!"}
      </p>
    </div>
  );
};

export default Status;
