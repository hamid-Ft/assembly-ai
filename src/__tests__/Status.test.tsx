import { render, screen } from '@testing-library/react';
import Status from '../components/Status';
import '@testing-library/jest-dom';

test('renders correct text when isLoading is true', () => {
    render(<Status isLoading status="uploading" />);
    const linkElement = screen.getByText(/Calculating...uploading.../i);
    expect(linkElement).toBeInTheDocument();
});

test('renders correct text when isLoading is false', () => {
    render(<Status isLoading={false} status="" />);
    const linkElement = screen.getByText(/Give me audio!/i);
    expect(linkElement).toBeInTheDocument();
});

test('matches snapshot when isLoading is true', () => {
    const { asFragment } = render(<Status isLoading status="uploading" />);
    expect(asFragment()).toMatchSnapshot();
});

test('matches snapshot when isLoading is false', () => {
    const { asFragment } = render(<Status isLoading={false} status="" />);
    expect(asFragment()).toMatchSnapshot();
});
