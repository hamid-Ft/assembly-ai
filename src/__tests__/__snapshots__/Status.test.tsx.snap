// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when isLoading is false 1`] = `
<DocumentFragment>
  <div>
    <p>
      Give me audio!
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when isLoading is true 1`] = `
<DocumentFragment>
  <div>
    <p>
      Calculating...uploading...
    </p>
  </div>
</DocumentFragment>
`;
